app/javascript/packs/components/Punches.js

```diff
@@ -2,11 +2,21 @@ import React from 'react';
 import Punch from './Punch';
 
 const Punches = ({ sheet }) => {
+
+  const worked_hours = (sheet) => {
```

Comment:
Nitpick: I believe the `worked_hours` function could be defined outside of the component (in the same file), since it doesn't depend on any component state.

It would also be better to name it `workedHours` for consistency with JS naming conventions.